aleo-programs/staking_lite/src/main.leo

```diff
@@ -0,0 +1,376 @@
+  }
+
+  finalize set_commission_percent(commission_rate: u128) {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 0u8);
+
+    // Make sure all commission is claimed before changing the rate
+    let balance: u64 = total_balance.get(0u8);
+    let bonded: u64 = 0u64; // credits.aleo/bonded.get(CORE_PROTOCOL);
+    assert_eq(balance, bonded);
+
+    commission_percent.set(0u8, commission_rate);
+  }
+
+  transition unbond_all(pool_balance: u64) {
+    assert_eq(self.caller, ADMIN);
+    credits.aleo/unbond_public(pool_balance);
+
+    return then finalize();
+  }
+
+  finalize unbond_all() {
+    // Make sure all commission is claimed before unbonding
+    let balance: u64 = total_balance.get(0u8);
+    let bonded: u64 = 0u64; // credits.aleo/bonded.get(CORE_PROTOCOL);
+    assert_eq(balance, bonded);
+
+    state.set(0u8, 1u8);
+  }
+
+  transition claim_all() {
+    assert_eq(self.caller, ADMIN);
+    credits.aleo/claim_unbond_public();
+
+    return then finalize();
+  }
+
+  finalize claim_all() {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 1u8);
+
+    state.set(0u8, 2u8);
+    current_batch_height.remove(0u8);
+  }
+
+  transition bond_all(validator_address: address, amount: u64) {
+    assert_eq(self.caller, ADMIN);
+    // Call will fail if there is any balance still bonded to another validator
+    credits.aleo/bond_public(validator_address, amount);
+
+    return then finalize(validator_address, amount);
+  }
+
+  finalize bond_all(validator_address: address, amount: u64) {
+    let current_state: u8 = state.get(0u8);
+    // Bonding allowed during normal operations or after full unbonding
+    assert(current_state == 0u8 || current_state == 2u8);
+
+    let unbonding_balance: u64 = 0u64; // credits.aleo/unbonding.get(CORE_PROTOCOL);
+    assert_eq(unbonding_balance, 0u64);
+
+    let account_balance: u64 = 0u64; // credits.aleo/account.get(CORE_PROTOCOL);
+    let pending_withdrawals: u64 = pending_withdrawal.get(0u8);
+    let available_balance: u64 = account_balance - pending_withdrawals;
+    assert(amount <= available_balance);
+
+    // Set validator and state in case this was part of a validator change
+    validator.set(0u8, validator_address);
+    state.set(0u8, 0u8);
+  }
+
+  transition claim_commission() {
+    assert_eq(self.caller, ADMIN);
+    return then finalize();
+  }
+
+  finalize claim_commission() {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 0u8);
+
+    // Distribute shares for new commission
+    let bonded: u64 = 0u64; // credits.aleo/bonded.get(CORE_PROTOCOL);
+    let current_balance: u64 = total_balance.get(0u8);
+    let current_shares: u64 = total_shares.get(0u8);
+    let rewards: u64 = bonded - current_balance;
+    let commission_rate: u128 = commission_percent.get(0u8);
+    let new_commission: u64 = rewards > 0u64 ? get_commission(rewards as u128, commission_rate) : 0u64;
```

Comment:
I don't think this is necessary. If rewards is 0, there's no commission to claim and no point in calling this function.